Projects/Dotmim.Sync.Core/CoreProvider.ApplyChanges.cs

```diff
@@ -58,33 +50,46 @@ public virtual async Task<(SyncContext, ChangesApplied)> ApplyChangesAsync(SyncC
                     // -----------------------------------------------------
                     if (!message.FromScope.IsNewScope)
                     {
-                        changeApplicationAction = this.ApplyChangesInternal(context, message, connection, applyTransaction, DmRowState.Deleted, changesApplied);
-
+                        // for delete we must go from Up to Down
```

Comment:
Why the loop through all tables is now placed here ?
I mean, before we had this loop in the `ApplyChangesInternal` method.
This code works correctly and no reason for rollback it, but I want to know why you have decide to make the loop here and not in the `ApplyChangesInternal` anymore ?

Reply:
Ok, I answer my own question, it's because of bug #118 ...
I'm currently reviewing this PR, I try to understand why we have the issue (that we should not have even in the last version )
Thx for the PR !

Reply:
Ok I've understand.
It's awful to see that I did not managed this scenario so far …

So thanks a lot for your PR @gb0o !! Everything seems correct, we are good to merge !

Reply:
@Mimetis sorry for the late reply. I see you figured why the changes was needed. In my case I got a constraint violation on a row not yet inserted because all table updates were applied before all tables inserts. When updates and inserts are handled together for each table and in the correct order every thing works. The code ended up more verbose with multiple outer loops. I could possibly improve it by getting ride of the outer loops and passing a list of operations to `ApplyChangesInternal ` such as new[] {SqlCmd.Update, SqlCmd.Insert} where SqlCmd is an new enum. There might already be such a enum.